File: go-sdk/pkg/models/operations/getinvitesid.go
package operations

import (
	"openapi/pkg/models/shared"
	"openapi/pkg/utils"
)

type GetInvitesIDPathParams struct {
	InviteID string `pathParam:"style=simple,explode=false,name=invite_id"`
}

type GetInvitesIDQueryParams struct {
	Fields []string `queryParam:"style=form,explode=false,name=fields"`
}

type GetInvitesIDRequest struct {
	Retries     *utils.RetryConfig
	PathParams  GetInvitesIDPathParams
	QueryParams GetInvitesIDQueryParams
}

type GetInvitesIDResponse struct {
	ClientError *shared.ClientError
	ContentType string
	Invite      *shared.Invite
	StatusCode  int64
}

File: ts-sdk/src/sdk/models/operations/getinvitesid.ts
import { SpeakeasyMetadata, SpeakeasyBase, RetryConfig } from "../../../internal/utils";
import * as shared from "../shared";



export class GetInvitesIdPathParams extends SpeakeasyBase {
  @SpeakeasyMetadata({ data: "pathParam, style=simple;explode=false;name=invite_id" })
  inviteId: string;
}


export class GetInvitesIdQueryParams extends SpeakeasyBase {
  @SpeakeasyMetadata({ data: "queryParam, style=form;explode=false;name=fields" })
  fields?: string[];
}


export class GetInvitesIdRequest extends SpeakeasyBase {
  @SpeakeasyMetadata()
  retries?: RetryConfig;

  @SpeakeasyMetadata()
  pathParams: GetInvitesIdPathParams;

  @SpeakeasyMetadata()
  queryParams: GetInvitesIdQueryParams;
}


export class GetInvitesIdResponse extends SpeakeasyBase {
  @SpeakeasyMetadata()
  clientError?: shared.ClientError;

  @SpeakeasyMetadata()
  contentType: string;

  @SpeakeasyMetadata()
  invite?: shared.Invite;

  @SpeakeasyMetadata()
  statusCode: number;
}
